### PR TITLE
docs(serializers): fix hyperlinks to `salt.serializers`

### PR DESCRIPTION
**Pull request by myii**
### What does this PR do?

While reviewing https://github.com/saltstack-formulas/telegraf-formula/pull/4, found that the link to `salt.serializers` ([serializer modules](https://docs.saltstack.com/en/latest/ref/serializers/all/index.html)) wasn't working at:

* [`salt.modules.slsutil.serialize`](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.slsutil.html#salt.modules.slsutil.serialize)

Found a couple of other places in the docs this wasn't working, either:

* [`salt.modules.slsutil.deserialize`](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.slsutil.html#salt.modules.slsutil.deserialize)
* [`salt.states.file.serialize`](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.serialize)

Tested the fix by building the docs locally.  Also found a couple of further documentation issues which were resolved in the other two commits.

### What issues does this PR fix or reference?

N/a.

### Tests written?

N/a, docs only.

### Commits signed with GPG?

Yes.

### CODE REVIEW

**Review by waynew — APPROVED** [review]

**Comment by waynew** [comment]
@myii Thanks for the fix! Would you also go ahead and add this to the `fixed` section in the CHANGELOG.md?

**Comment by myii** [comment]
> ... Would you also go ahead and add this to the `fixed` section in the CHANGELOG.md?

@waynew Would that be under `3000.1`?  Also, couldn't this end up leading to various merge conflicts across PRs as each change the same lines in the changelog?

**Comment by waynew** [comment]
@myii it *shouldn't* - we've added it to the .gitattributes file that should just merge it all. If that doesn't work we'll have to take a different approach. But also we've been lazy about asking people to update the changelog 😝

**Comment by myii** [comment]
@waynew OK, done.  I'll do this for #56237 as well.

File: CHANGELOG.md
Comment (waynew) [inline CHANGELOG.md]: Oooh, I think you need to rebase - the version above is `## Unreleased (Neon)`. The latest master should have `3000.1`, though given our current schedule this will probably actually end out in `## Unreleased - 3001 (Magnesium)`, which I don't think I added yet 🙃

**Review by waynew — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (waynew) [reply CHANGELOG.md]: https://github.com/saltstack/salt/commit/e3216db3e51737c534c9b2a0154242600eaad3c2 That comment was correct - I hadn't added the Magnesium release to the top of the changelog

Comment (myii) [reply CHANGELOG.md]: @waynew Done, apologies for the mix-up.

Comment (myii) [reply CHANGELOG.md]: @waynew I just noticed:

> ```
> ### Unreleased (3000.1)
> ```

Surely that should be:

```diff
-### Unreleased (3000.1)
+## Unreleased (3000.1)
```

Comment (waynew) [reply CHANGELOG.md]: 🤦‍♂ doh! My bad. Yes it should. We'll probably fix that with Sodium - the changelog is important, but I wouldn't say it's *critical* 😝

**Review by waynew — DISMISSED** [review]

**Comment by frogunder** [comment]
@myii 

Seems like there is a conflict, can you take a look, please?

**Comment by myii** [comment]
@frogunder https://github.com/saltstack/salt/pull/56325#issuecomment-597491868.

> @myii it _shouldn't_ - we've added it to the .gitattributes file that should just merge it all. If that doesn't work we'll have to take a different approach. But also we've been lazy about asking people to update the changelog stuck_out_tongue_closed_eyes

@waynew Conflicting files: `CHANGELOG.md`!  How would you like me to deal with this?

**Comment by waynew** [comment]
@myii I went ahead and resolved it. And fixed the `### 3000.1` -> `## 3000.1` thing 👍

**Comment by myii** [comment]
> @myii I went ahead and resolved it. And fixed the `### 3000.1` -> `## 3000.1` thing 👍

Thanks!

File: CHANGELOG.md
Comment (Ch3LL) [inline CHANGELOG.md:9]: We already released 3000.1 so this would make it into Sodium (30001)

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (myii) [reply CHANGELOG.md]: @Ch3LL Done, I had to prepare the section for `3001` as well.  I took the liberty to move my other entry to `3001` (#56237), since that was merged after the release of `3000.1` as well.

**Review by Ch3LL — APPROVED** [review]

**Comment by Ch3LL** [comment]
@waynew are you good with this? Just waiting on your approval here

**Review by waynew — APPROVED** [review]

**Comment by waynew** [comment]
@Ch3LL :rocket:

**Pull request merged**
